src/governance.cpp

```diff
@@ -954,23 +954,22 @@ void CGovernanceManager::RequestGovernanceObject(CNode* pfrom, const uint256& nH
         return;
     }
 
-    CBloomFilter filter(GOVERNANCE_FILTER_ELEMENTS, GOVERNANCE_FILTER_FP_RATE, GetRandInt(999999), BLOOM_UPDATE_ALL);
+    CBloomFilter filter;
+    filter.clear();
 
     if(fUseFilter) {
         CGovernanceObject* pObj = FindGovernanceObject(nHash);
 
         if(pObj) {
+            int nFilterElements = min(3*mnodeman.CountMasternodes(), GOVERNANCE_FILTER_MAX_ELEMENTS); 
```

Comment:
I don't think a lower bound would do much good here, but if for example the mn list is empty it may well cause problems.

I think it's best to revert to a fixed size.

I don't think it's worth making the size network dependent just for testnet and also that somewhat defeats the purpose of testnet testing.  For determining timings it's better to use the worse case I think.